aws/resource_aws_storagegateway_nfs_file_share.go

```diff
@@ -89,26 +97,28 @@ func resourceAwsStorageGatewayNfsFileShare() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"directory_mode": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "0777",
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "0777",
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[0-7]{4}$`), ""),
 						},
 						"file_mode": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "0666",
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "0666",
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[0-7]{4}$`), ""),
 						},
 						"group_id": {
 							Type:         schema.TypeInt,
 							Optional:     true,
 							Default:      65534,
-							ValidateFunc: validation.IntAtLeast(0),
+							ValidateFunc: validation.IntBetween(0, 4294967294),
```

Comment:
`aws/resource_aws_storagegateway_nfs_file_share.go:115:47: constant 4294967294 overflows int`
I think this is for 32-bit targets.
See a discussion [here](https://github.com/hashicorp/terraform-provider-aws/pull/14030#pullrequestreview-441689970) on resolution.

Reply:
Done. Thanks for this!